lib/block.mli

```diff
@@ -33,23 +83,36 @@ type t = {
   line : int;
   file : string;
   section : section option;
+  dir : string option;
+  source_trees : string list;
+  required_packages : string list;
   labels : Label.t list;
-  header : string option;
   contents : string list;
+  skip : bool;
+  version_enabled : bool;
+      (** Whether the current OCaml version complies with the block's version. *)
+  set_variables : (string * string) list;
+  unset_variables : string list;
   value : value;
```

Comment:
Sorry to bring the debate again but the `value` doesn't need to be in this record.

Some part of the program doesn't need the "value" to be computed (for example `Document` or the `dep` command).
Given the implementation of `Block.mk`, it should be easy to compute it afterwards (from `t`), avoiding unnecessary error checks when they are not necessary (it's only used in `bin/test/main.ml`).

Also, the `value` representation could evolve and doesn't need the restriction of being of being tied to the representation of a "parsed block" or the restriction of making sense in every part of the program.
I think it should be moved to its own module (for example `Evaluated_block`).

Reply:
Most of what is currently in the value type is not really a result of evaluation, maybe I need to rename it first. The result of evaluation is only the `tests` of `cram_value` and the `phrases` of `toplevel_value`.
Maybe we can try to split those from the rest.
But I'm not sure we can trim more things from the checks of `Block.mk` as the label checks sound like a good thing even if the block doesn't need to be evaluated. Do you agree?

Reply:
Some label checks doesn't need to be done if we don't need `value`.
What I mean is that `value` is something that will change as we change `bin/test/main` but the rest of `Block.t` have to make sense for every users of it (every other commands).

Reply:
I strongly disagree here, the whole point of this PR is not to process invalid mdx document and to unify the error handling. I agree `value` isn't the best suited name but so was `eval`.

Nothing is evaluated, `value` is just the cleaned up block. We don't ever want to deal with improper blocks outside of this module.

Of course the `dep` command need the blocks to be valid. We don't want to add a file or directory dependency if we shouldn't.